src/Nethermind/Nethermind.Blockchain/Synchronization/FastSync/NodeDataDownloader.cs

```
@@ -22,125 +22,69 @@
-                        _noAllocPunishment = true;
-                        if(_logger.IsInfo) _logger.Info("Adjusting useful peers");
-                        _noAllocInARow = 0;
                     }
-                }
-
-                (NodeDataHandlerResult Result, int NodesConsumed) result = (NodeDataHandlerResult.InvalidFormat, 0);
-                try
-                {
-                    result = _nodeDataFeed.HandleResponse(batch);
-                    if (result.Result == NodeDataHandlerResult.BadQuality)
-                    {
-                        _syncPeerPool.ReportBadPeer(batch.AssignedPeer);
-                    }
-                }
-                catch (Exception e)
-                {
-                    if (_logger.IsError) _logger.Error($"Error when handling response", e);
-                }
-
-                Interlocked.Add(ref _consumedNodesCount, result.NodesConsumed);
-                if (result.NodesConsumed == 0 && peer != null)
-                {
-                    _syncPeerPool.ReportNoSyncProgress(nodeSyncAllocation);
-                }
+                );
             }
-            finally
+
+            (NodeDataHandlerResult Result, int NodesConsumed) result = (NodeDataHandlerResult.InvalidFormat, 0);
+            try
             {
-                if (nodeSyncAllocation != null)
+                result = _feed.HandleResponse(batch);
+                if (result.Result == NodeDataHandlerResult.BadQuality)
                 {
-                    _syncPeerPool.Free(nodeSyncAllocation);
+                    _syncPeerPool.ReportBadPeer(batch.AssignedPeer);
                 }
             }
-        }
-
-        private async Task UpdateParallelism()
-        {
-            int newUsefulPeerCount = Math.Max(0, _syncPeerPool.UsefulPeerCount - (_noAllocPunishment ? 1 : 0));
-            int difference = newUsefulPeerCount - _lastUsefulPeerCount;
-            if (difference == 0)
+            catch (Exception e)
```

DO we really want to swallow all exceptions here?

---

yeah - this is unrecoverble error - it will display a lot of issues and then user is exepcted to close the app

---

But maybe we can narrow the exception types? If not its fine